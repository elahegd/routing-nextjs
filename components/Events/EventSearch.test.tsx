import React from "react";
import "@testing-library/jest-dom";
import { render, fireEvent } from "@testing-library/react";
import EventSearch from "./EventSearch";

describe("EventSearch", () => {
  it("renders form correctly", () => {
    const { getByTestId, getByText } = render(
      <EventSearch onSearch={() => {}} />
    );
    const yearInput = getByTestId("year");
    const monthInput = getByTestId("month");
    const submitButton = getByText("Find Events");
    expect(yearInput).toBeInTheDocument();
    expect(monthInput).toBeInTheDocument();
    expect(submitButton).toBeInTheDocument();
  });

  it("calls onSearch with correct values", () => {
    const onSearch = jest.fn();
    const { getByTestId, getByText } = render(
      <EventSearch onSearch={onSearch} />
    );
    const yearInput = getByTestId("year") as HTMLSelectElement;
    const monthInput = getByTestId("month") as HTMLSelectElement;
    const submitButton = getByText("Find Events");
    fireEvent.change(yearInput, { target: { value: "2022" } });
    fireEvent.change(monthInput, { target: { value: "2" } });
    fireEvent.click(submitButton);
    expect(onSearch).toHaveBeenCalledWith("2022", "2");
  });

  it("has correct class name", () => {
    const { container } = render(<EventSearch onSearch={() => {}} />);
    const formElement = container.querySelector("form");
    expect(formElement).toHaveClass("form");
  });

  it("matches snapshot", () => {
    const { container } = render(<EventSearch onSearch={() => {}} />);
    expect(container).toMatchSnapshot();
  });
});
