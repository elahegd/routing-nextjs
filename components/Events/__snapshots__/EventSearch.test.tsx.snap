// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventSearch matches snapshot 1`] = `
<div>
  <form
    class="form"
    name="filterEvents"
  >
    <div
      class="controls"
    >
      <div
        class="control"
      >
        <label
          for="year"
        >
          Year
        </label>
        <select
          data-testid="year"
          id="year"
        >
          <option
            value="2021"
          >
            2021
          </option>
          <option
            value="2022"
          >
            2022
          </option>
        </select>
      </div>
      <div
        class="control"
      >
        <label
          for="month"
        >
          Month
        </label>
        <select
          data-testid="month"
          id="month"
        >
          <option
            value="1"
          >
            January
          </option>
          <option
            value="2"
          >
            February
          </option>
          <option
            value="3"
          >
            March
          </option>
          <option
            value="4"
          >
            April
          </option>
          <option
            value="5"
          >
            May
          </option>
          <option
            value="6"
          >
            June
          </option>
          <option
            value="7"
          >
            July
          </option>
          <option
            value="8"
          >
            August
          </option>
          <option
            value="9"
          >
            September
          </option>
          <option
            value="10"
          >
            October
          </option>
          <option
            value="11"
          >
            November
          </option>
          <option
            value="12"
          >
            December
          </option>
        </select>
      </div>
    </div>
    <button
      class="btn"
    >
      Find Events
    </button>
  </form>
</div>
`;
